presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -115,132 +72,41 @@ public SliceDictionaryColumnWriter(
     }
 
-    @Override
-    public int getIndexBytes()
-    {
-        checkState(!directEncoded);
-        return toIntExact(estimateIndexBytesPerValue(dictionary.getEntryCount()) * getNonNullValueCount());
-    }
-
-    @Override
-    public long getValueCount()
-    {
-        checkState(!directEncoded);
-        return totalValueCount;
-    }
-
-    @Override
-    public long getNonNullValueCount()
-    {
-        checkState(!directEncoded);
-        return totalNonNullValueCount;
-    }
-
     @Override
     public int getDictionaryEntries()
     {
-        checkState(!directEncoded);
+        checkState(!isDirectEncoded());
         return dictionary.getEntryCount();
     }
 
     @Override
-    public OptionalInt tryConvertToDirect(int maxDirectBytes)
+    protected int getInstanceSize()
     {
-        checkState(!closed);
-        checkState(!directEncoded);
-        if (directColumnWriter == null) {
-            directColumnWriter = new SliceDirectColumnWriter(column, type, compressionParameters, dwrfEncryptor, orcEncoding, this::newStringStatisticsBuilder, metadataWriter);
-        }
-        checkState(directColumnWriter.getBufferedBytes() == 0);
-
-        Block dictionaryValues = dictionary.getElementBlock();
-        for (DictionaryRowGroup rowGroup : rowGroups) {
-            directColumnWriter.beginRowGroup();
-            // todo we should be able to pass the stats down to avoid recalculating min and max
-            boolean success = writeDictionaryRowGroup(dictionaryValues, rowGroup.getValueCount(), rowGroup.getDictionaryIndexes(), maxDirectBytes);
-            directColumnWriter.finishRowGroup();
-
-            if (!success) {
-                directColumnWriter.close();
-                directColumnWriter.reset();
-                return OptionalInt.empty();
-            }
-        }
-
-        if (inRowGroup) {
-            directColumnWriter.beginRowGroup();
-            if (!writeDictionaryRowGroup(dictionaryValues, rowGroupValueCount, values, maxDirectBytes)) {
-                directColumnWriter.close();
-                directColumnWriter.reset();
-                return OptionalInt.empty();
-            }
-        }
-        else {
-            checkState(rowGroupValueCount == 0);
-        }
-
-        rowGroups.clear();
-
-        // free the dictionary
-        dictionary.clear();
-
-        rawBytes = 0;
-        totalValueCount = 0;
-        totalNonNullValueCount = 0;
-
-        rowGroupValueCount = 0;
-        statisticsBuilder = newStringStatisticsBuilder();
-
-        directEncoded = true;
-
-        return OptionalInt.of(toIntExact(directColumnWriter.getBufferedBytes()));
+        return INSTANCE_SIZE;
     }
 
-    private boolean writeDictionaryRowGroup(Block dictionary, int valueCount, IntBigArray dictionaryIndexes, int maxDirectBytes)
+    @Override
+    protected boolean tryConvertToDirect(int rowGroupValueCount, IntBigArray rowGroupValues, int maxDirectBytes)
```

Comment:
Why did we change the name from `dictionaryIndexes` to `rowGroupValues`? By looking at the logic, it seems to be dictionaryIndexes?